### DISCUSSION TITLE
How to skip lsp auto format  for html <pre> tag?

### DISCUSSION THREAD

**Discussion by tonmoy998** [LazyVim/LazyVim#1335] [Q&A] [answered]
I working with asci code in html.Each time I save auto format changes asci code.

**Comment by dpetka2001** [comment, marked as answer]
`<leader>uf` to toggle it off temporarily.

**Reply by tonmoy998** [reply]
Thank you.In addition can I enable it for only pre tag?

**Reply by dpetka2001** [reply]
The `vim.lsp.buf.format` is for the whole buffer as far as I'm aware. You could specify for specific LSP client if you have multiple clients attached to the buffer. I also know that you can visually select ranges and then `<leader>cf` to only format that range. But I'm not sure if you can exclude specific ranges from formatting. Maybe someone who is more informed than me can give a more concrete solution.

**Reply by tonmoy998** [reply]
ok

**Comment by jasalt** [comment]
Testing this while editing PHP files with HTML and "Format on save" (triggered from plugins/lsp/format.lua) does not seem to touch `<pre>` contents while using intelephense. `=` indent command however re-indents lines inside `<pre>` into same column (indented one level deeper). I guess getting those two separate indenting methods working consistently is work in progress as it's mentioned somewhere in treesitter related docs. This `<leader>cf` indenting command was new to me and seems useful in place of `=` for now eventhough it's non-standard.

[tonmoy998](https://github.com/tonmoy998) I guess it would be better to move such code into a separate template partial, simply loading/importing it into main template and then avoiding of touching it with editor having intelligent features on that can mess with it.

I noted some flakiness while I was testing and answering to this where the "Format on save" started eating up all new code that was written after I restarted LSP. It was also indenting stuff inside existing `<pre>`'s in weird ways.. argh. Restart fixed it.

**Reply by tonmoy998** [reply]
Thanks for your suggestions.Even I feel the same
